enclave_build/src/cache.rs

```diff
@@ -143,6 +146,102 @@ impl CacheManager {
         })
     }
 
+    /// Stores the image data provided as argument in the cache at the folder pointed
+    /// by the 'root_path' field.
+    pub fn store_image_data(&mut self, image_name: &str, image_data: &ImageData) -> Result<()> {
+        // Create the folder where the image data will be stored. Each image blob will be stored in
+        // a file named by the SHA256 digest of the content.
+        let blobs_path = self.root_path.join(CACHE_BLOBS_FOLDER);
+        fs::create_dir_all(&blobs_path).map_err(EnclaveBuildError::CacheStoreError)?;
+
+        for layer in &image_data.layers {
+            // Each layer file will be named after the layer's digest hash
+            let layer_file_path =
+                blobs_path.join(format!("{:x}", sha2::Sha256::digest(&layer.data)));
+            File::create(&layer_file_path)
+                .map_err(EnclaveBuildError::CacheStoreError)?
+                .write_all(&layer.data)
+                .map_err(EnclaveBuildError::CacheStoreError)?;
+        }
+
+        // Store the manifest
+        let manifest = image_data
+            .manifest
+            .as_ref()
+            .ok_or(EnclaveBuildError::ManifestError)?;
+        let manifest_bytes = serde_json::to_vec(manifest).map_err(EnclaveBuildError::SerdeError)?;
+
+        File::create(&blobs_path.join(format!("{:x}", sha2::Sha256::digest(&manifest_bytes))))
+            .map_err(EnclaveBuildError::CacheStoreError)?
+            .write_all(&manifest_bytes)
+            .map_err(EnclaveBuildError::CacheStoreError)?;
+
+        // Store the config and validate UTF8 bytes
+        let config_json = String::from_utf8(image_data.config.data.clone()).map_err(|_| {
+            EnclaveBuildError::CacheStoreError(Error::new(
+                ErrorKind::InvalidData,
+                "Config data invalid",
+            ))
+        })?;
+        let config_digest = format!("{:x}", sha2::Sha256::digest(config_json.as_bytes()));
+
+        File::create(&blobs_path.join(&config_digest))
+            .map_err(EnclaveBuildError::CacheStoreError)?
+            .write_all(config_json.as_bytes())
+            .map_err(EnclaveBuildError::CacheStoreError)?;
+
+        // If index file present, read and append new image entry
+        let index_json: CacheIndex = match File::options().read(true).open(CACHE_INDEX_FILE_NAME) {
```

Comment:
Again I think you can use File::open, also I think you can make the `match` a bit more Rusty, by using `map_or_else` or something else.